Dockerfile

```diff
@@ -24,22 +25,43 @@ RUN mix deps.get --only $MIX_ENV
 RUN mix deps.compile
 
 # build assets
+COPY priv priv
+
+# Note: if your project uses a tool like https://purgecss.com/,
+# which customizes asset compilation based on what it finds in
+# your Elixir templates, you will need to move the asset compilation step
+# down so that `lib` is available.
 COPY assets assets
+# webpack docker config needs to replace the regular webpack config due to env vars
 RUN rm assets/webpack.config.js
 RUN mv assets/webpack-docker.config.js assets/webpack.config.js
+# use webpack to compile npm dependencies - https://www.npmjs.com/package/webpack-deploy
 RUN cd assets && yarn && yarn run deploy
+# COPY package.json package-lock.json ./assets/
 RUN mix phx.digest
 
-# build project
-COPY priv priv
+# compile and build the release
 COPY lib lib
 RUN mix compile
-
+# changes to config/releases.exs don't require recompiling the code
+COPY config/releases.exs config/
+# uncomment COPY if rel/ exists
+# COPY rel rel
 RUN mix release
 
-EXPOSE 4000
-COPY entrypoint.sh .
-RUN chown -R nobody: /app
+# Start a new build stage so that the final image will only contain
+# the compiled release and other runtime necessities
+FROM alpine:3.12.1 AS app
+RUN apk add --no-cache libstdc++ openssl ncurses-libs
+
+ARG MIX_ENV
+ENV USER="nobody"
+
+WORKDIR "/home/nobody/app"
+
+COPY --from=build --chown="${USER}" /app/_build/prod/rel/console ./
+RUN chown -R nobody: .
```

Comment:
is this still needed if you're copying as `nobody`?

Reply:
it builds ok without it but then I get tons of permission errors when trying to run console. was a pain to figure out but this doesn't take nearly as long as before.